src/components/ViewAgents/SearchBarWizards.tsx

```diff
@@ -0,0 +1,133 @@
+        setSearchButtonActive: React.Dispatch<React.SetStateAction<boolean>>
+        suggestionResults: WizardDetailsBasicWithTimeStamp[]
+        setSuggestionResults: React.Dispatch<React.SetStateAction<Array<WizardDetailsBasicWithTimeStamp>>>
+    }
+) {
+
+    const navigate = useNavigate();
+    
+    const [suggestionActive, setSuggestionActive] = useState(false);
+    const [suggestionIndex, setSuggestionIndex] = useState<number>(-1);
+    const searchQueryRef = useRef<HTMLInputElement>(null);
+
+    function searchWizards(
+        popularWizards: WizardDetailsBasicWithTimeStamp[],
+        searchQuery: string
+    ) {
+        if (popularWizards === undefined) return undefined;
+        if (searchQuery === "") {
+          setSuggestionResults([]);
+          return undefined
+        }
+    
+        const results = popularWizards.filter(agent =>agent.name.toLowerCase().includes(searchQuery.toLowerCase().trim()));
+    
+        setSuggestionResults(results);
+    };
+
+    function inputOnchange(e: React.ChangeEvent<HTMLInputElement>) {
+      if(e.target.value === "") {
+        setSearchButtonActive(false);
+      }
+      setSuggestionActive(true)
+      searchWizards(popularWizards, e.target.value)
+    }
+
+    function inputOnFocus() {
+      if(searchQueryRef.current === null) return;
+      if(searchQueryRef.current.value === "" && suggestionActive) setSuggestionActive(false);
+    }
+    
+    function inputOnKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
+      if(searchQueryRef.current === null || searchQueryRef.current?.value === "") return;
+      if(!suggestionActive) return;
+      if (e.key === 'Enter') {
+          if(suggestionIndex !== -1) navigate(`/chat/${suggestionResults[suggestionIndex]?.id}`);
+          else {
+              setSearchButtonActive(true);
+              setSuggestionActive(false);
+          }
+      }
+      if (e.key === 'ArrowDown') {
+          setSuggestionIndex(prev => Number(prev) + 1)
+      }
+      if (e.key === 'ArrowUp') {
+          setSuggestionIndex(prev => Number(prev) - 1)
+      }
+    }
+
+    function searchButtonOnClick() {
+      if(searchQueryRef.current === null) return;
+      setSearchButtonActive(true);
+      setSuggestionActive(false);
+    }
+
+  return (
+        <>
+          <div className="position-relative w-100 max-w-500p d-flex align-items-center">
+            <Form.Control
+              onChange={e => inputOnchange(e)}
+              onKeyDown={(e) => inputOnKeyDown(e)}
+              onFocus={() => inputOnFocus()}
+              onBlur={() => setTimeout(() => setSuggestionActive(false), 50)}
+              onBeforeInput={() => setSearchButtonActive(false)}
+              style={{ color: "#fff" }}
+              placeholder="Agent Search"
+              aria-label="Agent Search"
+              ref={searchQueryRef}
+            />
+            {
+              !!searchQueryRef.current?.value && searchQueryRef.current?.value.length > 0 &&
```

Comment:
we just require this condition here right, both checks are the same thing, right?
```suggestion
              !!searchQueryRef.current?.value && 
```

Reply:
I had actually made this change locally yesterday, checking it out

Reply:
https://github.com/elna-ai/ELNA-DApp/pull/204/commits/7888d39dd7acb2b4b2f4d08148e3161aa6150055